clicommand/artifact_download.go

```diff
@@ -22,17 +22,22 @@ Description:
 
 Example:
 
-   $ buildkite-agent artifact download "pkg/*.tar.gz" . --build xxx
+   $ buildkite-agent artifact download --build xxx "pkg/*.tar.gz" .
 
    This will search across all the artifacts for the build with files that match that part.
    The first argument is the search query, and the second argument is the download destination.
 
    If you're trying to download a specific file, and there are multiple artifacts from different
    jobs, you can target the particular job you want to download the artifact from:
 
-   $ buildkite-agent artifact download "pkg/*.tar.gz" . --step "tests" --build xxx
+   $ buildkite-agent artifact download --step "tests" --build xxx "pkg/*.tar.gz" .
 
-   You can also use the step's jobs id (provided by the environment variable $BUILDKITE_JOB_ID)`
+   You can also use the step's job id (provided by the environment variable $BUILDKITE_JOB_ID)
+
+   By default, only artifacts from the most recent job in a chain of retried jobs are downloaded.
+   To include artifacts from previous retried jobs, use the "--include-retried-jobs" flag:
```

Comment:
Yeah, I agree. How about

```suggestion
   To include artifacts also from all retried jobs belonging to this step, use the "--include-retried-jobs" flag:
```

"Also", to make clear that it is as well as the last one. What do you reckon?

Reply:
CC @ticky

Reply:
I’m not sure this solves for the ambiguity in “retried,” and do we otherwise expose the step/job dichotomy to users?

Reply:
Good point. I'll ponder and reword tomorrow.

Reply:
We expose step just a couple of lines above, on line 35. But if we're not specifying a job ID, we're just using the artifact path, right, so this flag is just saying "get all the stuff, not just the most recent stuff", so mentioning the step isn't helpful. This might be better:

```suggestion
   To include matching artifacts also from previous retried jobs as well as the most recent, use the "--include-retried-jobs" flag:
```